mslib/utils/config.py

```diff
@@ -0,0 +1,299 @@
+dict_option_structure = {
+    "predefined_map_sections": {
+        "new_map_section": {
+            "CRS": "crs_value",
+            "map": {
+                "llcrnrlon": 0.0,
+                "llcrnrlat": 0.0,
+                "urcrnrlon": 0.0,
+                "urcrnrlat": 0.0,
+            },
+        }
+    },
+    "MSC_login": {
+        "http://www.your-mscolab-server.de": ["yourusername", "yourpassword"],
+    },
+    "WMS_login": {
+        "http://www.your-wms-server.de": ["yourusername", "yourpassword"],
+    },
+    "locations": {
+        "new-location": [0.0, 0.0],
+    },
+    "export_plugins": {
+        "plugin-name": ["extension", "module", "function", "default"],
+    },
+    "import_plugins": {
+        "plugin-name": ["extension", "module", "function", "default"],
+    },
+    "proxies": {
+        "https": "https://proxy.com",
+    },
+}
+
+# List options with predefined structure
+list_option_structure = {
+    "default_WMS": ["https://wms-server-url.com"],
+    "default_VSEC_WMS": ["https://vsec-wms-server-url.com"],
+    "default_LSEC_WMS": ["https://lsec-wms-server-url.com"],
+    "default_MSCOLAB": ["https://mscolab-server-url.com"],
+    "new_flighttrack_template": ["new-location"],
+    "gravatar_ids": ["example@email.com"],
+    "WMS_preload": ["https://wms-preload-url.com"],
+}
+
+
+def dict_raise_on_duplicates_empty(ordered_pairs):
+    """Reject duplicate and empty keys."""
+    accepted = {}
+    for key, value in ordered_pairs:
+        if key in accepted:
+            raise ValueError(f"duplicate key found: {key}")
+        elif key == "":
+            raise ValueError("empty key found")
+        else:
+            accepted[key] = value
+    return accepted
+
+
+def read_config_file(path=constants.MSS_SETTINGS):
+    """
+    reads a config file
+
+    Args:
+        config_file: name of config file
+
+    Returns: a dictionary
+    """
+    # path = constants.MSS_SETTINGS
+    path = path.replace("\\", "/")
+    dir_name, file_name = fs.path.split(path)
+    json_file_data = {}
+    error_message = ""
+    with fs.open_fs(dir_name) as _fs:
+        if _fs.exists(file_name):
+            file_content = _fs.readtext(file_name)
+            try:
+                json_file_data = json.loads(file_content, object_pairs_hook=dict_raise_on_duplicates_empty)
+            except json.JSONDecodeError as e:
+                logging.error(f"Error while loading json file {e}")
+                error_message = f"Unexpected error while loading config\n{e}"
+                raise FatalUserError(error_message)
+            except ValueError as e:
+                logging.error(f"Error while loading json file {e}")
+                error_message = f"Invalid keys detected in config\n{e}"
+                raise FatalUserError(error_message)
+        else:
+            error_message = f"MSS config File '{path}' not found"
+            raise FatalUserError(error_message)
+
+    global user_options
```

Comment:
with global I'm very carefully and I had already a lot of the problems here
https://stackoverflow.com/questions/19158339/why-are-global-variables-evil

Reply:
I'll look into this

Reply:
evil is a strong word. I think the use is justified here. Please not that you do not need it here, actually, as user_options is a dictionary, which you might modify in-place. 
E.g.
`
copied_config = copy.deepcopy(user_options)
user_options.clear()
user_config.update(copied_config)
`
or similar.

Reply:
Actually, many arguments against global variables can be made against classes and in Python the distinction between a class and a module is marginal.
"State" in any form is "evil", IMHO, but functional programming languages (e.g. Haskell) lacking state have somehow problems of not being adopted due to being very difficult to work with. But man, is the code easily verifiable and testable.